src/core/shiftManagement.ts

```diff
@@ -55,3 +62,47 @@ export function applyToShift(
     },
   });
 }
+
+export function rateWorker(
+  workerUuid: Worker["uuid"],
+  shiftUuid: Shift["uuid"],
+  rating: number,
+): Promise<ShiftAssignment> {
+  return prisma.shiftAssignment.update({
+    where: {
+      shiftUuid_workerUuid: {
+        shiftUuid,
+        workerUuid,
+      },
+    },
+    data: { rating },
+  });
+}
+
+export async function blockWorker(
+  workerUuid: Worker["uuid"],
+  shiftUuid: Shift["uuid"],
+  blockReason: string,
```

Comment:
Perform validation for blockReason. Check with business if this is mandatory field

Reply:
This is a free form text and a part of the requirement. Not sure what validation is needed here, may be some sanitization? Anyway, not part of my review comment.